src/Market.ts

```diff
@@ -33,14 +33,20 @@ export class Market implements MarketProperties {
     return cost;
   }
 
-  getResourceCostForOpenSlots(openSlots: number): number {
-    if (openSlots > this.maxOpenSlots) {
-      openSlots = this.maxOpenSlots;
+  getResourceBenefit(count: number): number {
+    let benefit = 0;
+    if (count > this.openSlots) {count = this.openSlots};
+    if (this.openSlots === 0 ) {benefit = 0} else {
```

Comment:
Same about the if statements

Reply:
Also, you can reduce the complexity of the code by returning early:

```ts
if (this.openSlots === 0) {
  return 0;
}
...
```
which is easier to read than
```ts
if (...) {
  ...
} else {
  ...
}
return ...;

Reply:
Also, the check should be on `count` technically. It's equivalent right now, but it's more straightforward to check the `count` variable.

Reply:
Also, you don't need to do this check, because the for loop won't run anyway if `count` is 0